presto-main/src/main/java/com/facebook/presto/execution/scheduler/NodeScheduler.java

```diff
@@ -67,6 +68,7 @@
     private final List<CounterStat> topologicalSplitCounters;
     private final List<String> networkLocationSegmentNames;
     private final InternalNodeManager nodeManager;
+    private final NodeSelectionStats nodeSelectionStats;
```

Comment:
Would you need a getter for this class to ensure it could be exposed to jmx?

Reply:
There are getters inside NodeSelectionStats, which would be exposed to JMX. I have tested on our cluster.